### PR TITLE
Fix build on latest nightly

### PR DESCRIPTION
**Pull request by Amanieu**
asm! is now stable.

### CODE REVIEW

File: src/elision.rs [hunk 1]
```diff
@@ -5,6 +5,8 @@
 // http://opensource.org/licenses/MIT>, at your option. This file may not be
 // copied, modified, or distributed except according to those terms.
 
+#[cfg(all(feature = "nightly", any(target_arch = "x86", target_arch = "x86_64")))]
```
Comment (bjorn3) [inline src/elision.rs:8]: Maybe rename the feature to "hardware-lock-elision" now or in a future release?

Comment (Amanieu) [reply src/elision.rs:8]: In a future version I will just have this always enabled by default.

Comment (bjorn3) [reply src/elision.rs:8]: Don't some processors have a bug causing "unpredictable system behavior" (aka crashes or worse) when using hardware lock elision?

Comment (Amanieu) [reply src/elision.rs:8]: That's actually a good point, I wasn't aware that HLE was broken on all currently available CPUs.

**Pull request merged**